src/server.cpp

```diff
@@ -2656,6 +2684,11 @@ void Server::DeleteClient(session_t peer_id, ClientDeletionReason reason)
 				++i;
 		}
 
+		// clear formspec info so the next client can't abuse the current state
+		auto formspec_info_iterator = m_formspec_state_data.find(peer_id);
+		if (formspec_info_iterator != m_formspec_state_data.end())
+			m_formspec_state_data.erase(formspec_info_iterator);
```

Comment:
`m_formspec_state_data.erase(peer_id);` is way shorter and does the same.

Reply:
oh ty, I wasn't aware it could be used even if the the key doesn't exist.